File: sdk/filegate/activity.go
package filegate

import (
	"context"
	"net/http"
	"net/url"
	"strconv"
)

type ActivityClient struct {
	core *clientCore
}

type ActivityListOptions struct {
	Limit     int
	Offset    int
	Query     string
	Operation string
	Outcome   string
}

func (c ActivityClient) List(ctx context.Context, opts ActivityListOptions) (*ActivityListResponse, error) {
	q := url.Values{}
	if opts.Limit > 0 {
		q.Set("limit", strconv.Itoa(opts.Limit))
	}
	if opts.Offset > 0 {
		q.Set("offset", strconv.Itoa(opts.Offset))
	}
	if opts.Query != "" {
		q.Set("q", opts.Query)
	}
	if opts.Operation != "" {
		q.Set("operation", opts.Operation)
	}
	if opts.Outcome != "" {
		q.Set("outcome", opts.Outcome)
	}
	var out ActivityListResponse
	if err := c.core.doJSON(ctx, http.MethodGet, "/v1/activity", q, nil, "", &out); err != nil {
		return nil, err
	}
	return &out, nil
}

File: sdk/ts/src/activity.ts
import { ClientCore } from "./core.js";
import type { ActivityListResponse } from "./types.js";

export type ActivityListConfig = {
  limit?: number;
  offset?: number;
  q?: string;
  operation?: string;
  outcome?: string;
};

export class ActivityClient {
  constructor(private readonly core: ClientCore) {}

  async list(config: ActivityListConfig = {}): Promise<ActivityListResponse> {
    const query: Record<string, string> = {};
    if (config.limit !== undefined) query.limit = String(config.limit);
    if (config.offset !== undefined) query.offset = String(config.offset);
    if (config.q) query.q = config.q;
    if (config.operation) query.operation = config.operation;
    if (config.outcome) query.outcome = config.outcome;
    return this.core.doJSON<ActivityListResponse>("GET", "/v1/activity", query);
  }
}
